app/services/proxy_status/dhcp.rb

```diff
@@ -0,0 +1,20 @@
+module ProxyStatus
+  class DHCP < Base
+
+    def subnets
+      return @subnets if @subnets
+
+      fetch_proxy_data do
+        @subnets = api.subnets.map do |subnet|
+          records = api.subnet(subnet['network'])
```

Comment:
It is a possibility, seems to me a little less intrusive than creating a new action directly in the DHCP module but we will be adding stuff into proxy's API nevertheless, so I think @witlessbird can tell us if it is a good idea.